route53/records.go

```diff
@@ -63,6 +63,43 @@ func (r *Route53) GetRecord(ctx context.Context, zoneID, host, recordType string
 	return nil, apierror.New(apierror.ErrNotFound, msg, nil)
 }
 
+// GetRecordByName gets a route53 resource record by name and by type if one is specified
+func (r *Route53) GetRecordByName(ctx context.Context, zoneID, name, recordType string) (*route53.ResourceRecordSet, error) {
```

Comment:
This function seems to be doing the same thing as `GetRecord` above (but more efficient). Do we need to keep both?

Reply:
true, good catch 💀 code